src/backend/bigquery/bigquery.test.ts

```diff
@@ -1,60 +1,231 @@
+    areaToSearch: SearchArea.Any,
+    caseSensitive: false,
+    ...options,
+  })
+  it('Calls the appropriate queries with default search type', async () => {
+    jest.spyOn(buildSqlQuery, 'buildSqlQuery').mockReturnValue('query')
+    ;(BigQuery.prototype.query as jest.Mock)
+      .mockResolvedValueOnce([
+        [{ locale: 'fr' }, { locale: 'en' }, { locale: 'de' }],
+      ])
+      .mockResolvedValueOnce([
+        [{ name: 'taxon1' }, { name: 'taxon2' }, { name: 'taxon3' }],
+      ])
+    await sendSearchQuery(makeSearchParams())
+    expect(BigQuery.prototype.query).toHaveBeenCalledTimes(2)
+    expect(BigQuery.prototype.query).toHaveBeenNthCalledWith(1, {
+      query: 'query',
+      location: 'europe-west2',
+      params: {
+        excluded_keyword0: 'excluded1',
+        excluded_keyword1: 'excluded2',
+        keyword0: 'keyword1',
+        keyword1: 'keyword2',
+        link: 'link',
+        locale: 'en',
+        organisation: 'organisation',
+        taxon: 'taxon',
+      },
+    })
+    expect(BigQuery.prototype.query).toHaveBeenNthCalledWith(2, {
+      query: `SELECT *
+         FROM search.thing
+         WHERE CONTAINS_SUBSTR(name, @selected_words_without_quotes)
+         ;`,
+      location: 'europe-west2',
+      params: {
+        selected_words_without_quotes: 'keyword1 keyword2',
+      },
+    })
+  })
+  it('Calls the appropriate queries with Taxon search type', async () => {
+    jest.spyOn(buildSqlQuery, 'buildSqlQuery').mockReturnValue('query')
+    ;(BigQuery.prototype.query as jest.Mock)
+      .mockResolvedValueOnce([
+        [{ locale: 'fr' }, { locale: 'en' }, { locale: 'de' }],
+      ])
+      .mockResolvedValueOnce([
+        [{ name: 'taxon1' }, { name: 'taxon2' }, { name: 'taxon3' }],
+      ])
+    await sendSearchQuery(makeSearchParams({ searchType: SearchType.Taxon }))
+    expect(BigQuery.prototype.query).toHaveBeenCalledTimes(2)
+    expect(BigQuery.prototype.query).toHaveBeenNthCalledWith(1, {
+      query: 'query',
+      location: 'europe-west2',
+      params: {
+        excluded_keyword0: 'excluded1',
+        excluded_keyword1: 'excluded2',
+        keyword0: 'keyword1',
+        keyword1: 'keyword2',
+        link: 'link',
+        locale: 'en',
+        organisation: 'organisation',
+        taxon: 'taxon',
+      },
+    })
+    expect(BigQuery.prototype.query).toHaveBeenNthCalledWith(2, {
+      query: `SELECT "Taxon" as type, * FROM search.taxon WHERE lower(name) = lower(@name);`,
+      location: 'europe-west2',
+      params: { name: 'taxon' },
+    })
+  })
+  it('Calls the appropriate queries with Organisation search type', async () => {
+    jest.spyOn(buildSqlQuery, 'buildSqlQuery').mockReturnValue('query')
+    ;(BigQuery.prototype.query as jest.Mock)
+      .mockResolvedValueOnce([
+        [{ locale: 'fr' }, { locale: 'en' }, { locale: 'de' }],
+      ])
+      .mockResolvedValueOnce([
+        [{ name: 'taxon1' }, { name: 'taxon2' }, { name: 'taxon3' }],
```

Comment:
Are these required here?  They aren't representative of the actual query results.